dist/index.d.ts

```diff
@@ -1949,87 +1949,55 @@ type EVENT_HEAL_TYPE_RANGED = 2;
 
 type EventDestroyType = "creep" | StructureConstant;
 
-type EventItem =
-    | {
-          type: EVENT_ATTACK;
-          objectId: string;
-          data: {
-              targetId: string;
-              damage: number;
-              attackType: EventAttackType;
-          };
-      }
-    | {
-          type: EVENT_OBJECT_DESTROYED;
-          objectId: string;
-          data: {
-              type: EventDestroyType;
-          };
-      }
-    | {
-          type: EVENT_ATTACK_CONTROLLER;
-          objectId: string;
-      }
-    | {
-          type: EVENT_BUILD;
-          objectId: string;
-          data: {
-              targetId: string;
-              amount: number;
-              energySpent: number;
-          };
-      }
-    | {
-          type: EVENT_HARVEST;
-          objectId: string;
-          data: {
-              targetId: string;
-              amount: number;
-          };
-      }
-    | {
-          type: EVENT_HEAL;
-          objectId: string;
-          data: {
-              targetId: string;
-              amount: number;
-              healType: EventHealType;
-          };
-      }
-    | {
-          type: EVENT_REPAIR;
-          objectId: string;
-          data: {
-              targetId: string;
-              amount: number;
-              energySpent: number;
-          };
-      }
-    | {
-          type: EVENT_RESERVE_CONTROLLER;
-          objectId: string;
-          data: {
-              amount: number;
-          };
-      }
-    | {
-          type: EVENT_UPGRADE_CONTROLLER;
-          objectId: string;
-          data:
-              | {
-                    amount: number;
-                    energySpent: number;
-                }
-              | {
-                    type: EVENT_EXIT;
-                    objectId: string;
-                    data: {
-                        room: string;
-                        x: number;
-                        y: number;
-                    };
-                };
-      };
+interface EventItem<T extends EventConstant = EventConstant> {
+    event: T;
+    objectId: string;
+    data: EventData[T];
+}
+
+interface EventData {
+    [key: number]: null | object;
```

Comment:
```suggestion
    [key: number]: null | any;
```
Type `object` has no properties. [Playground](http://www.typescriptlang.org/play/#src=interface%20With_Object%20%7B%0D%0A%20%20%20%20%5Bkey%3A%20number%5D%3A%20null%20%7C%20object%0D%0A%7D%0D%0Alet%20with_object%3A%20With_Object%20%3D%20%7B%7D%3B%0D%0A%0D%0Ainterface%20With_Any%20%7B%0D%0A%20%20%20%20%5Bkey%3A%20number%5D%3A%20null%20%7C%20any%0D%0A%7D%0D%0Alet%20with_any%3A%20With_Any%20%3D%20%7B%7D%3B%0D%0A%0D%0Atype%20Foo%20%3D%20%7B%20foo%3A%20string%20%7D%0D%0A%0D%0Awith_object%5B42%5D%20%3D%20%7B%20foo%3A%20%22foo%22%20%7D%0D%0A%0D%0Alet%20foo%3A%20Foo%20%3D%20with_object%5B42%5D%3B%0D%0A%0D%0Awith_any%5B42%5D%20%3D%20%7B%20foo%3A%20%22bar%22%20%7D%0D%0A%0D%0Alet%20bar%3A%20Foo%20%3D%20with_any%5B42%5D%3B%0D%0A%0D%0Aif%20(with_object%5B42%5D.property%20%3D%3D%3D%20undefined)%20%7B%0D%0A%20%20%20%20%2F%2F...%0D%0A%7D%0D%0A%0D%0Aif%20(with_any%5B42%5D.property%20%3D%3D%3D%20undefined)%20%7B%0D%0A%20%20%20%20%2F%2F...%0D%0A%7D)